src/commands/trace/trace.ts

```diff
@@ -0,0 +1,127 @@
+import chalk from 'chalk'
+import {spawnSync} from 'child_process'
+import {Command} from 'clipanion'
+import crypto from 'crypto'
+import os from 'os'
+import {parseTags} from '../../helpers/tags'
+import {apiConstructor} from './api'
+import {APIHelper} from './interfaces'
+
+export class TraceCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Trace a command with a custom span and report it to Datadog.',
+    details: `
+            This command wraps another command, which it will launch, and report a custom span to Datadog.
+            See README for details.
+        `,
+    examples: [
+      ['Trace a command and report to Datadog', 'datadog-ci trace --name "Say Hello" -- echo "Hello World"'],
+      [
+        'Trace a command and report to the datadoghq.eu site',
+        'DATADOG_SITE=datadoghq.eu datadog-ci trace -- echo "Hello World"',
+      ],
+    ],
+  })
+  private command?: string[]
+
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY || process.env.DD_API_KEY,
+    envVarTags: process.env.DD_TAGS,
+  }
+
+  private name?: string
+
+  public async execute() {
+    if (!this.command || !this.command.length) {
+      this.context.stderr.write('Missing command to run\n')
+
+      return 1
+    }
+
+    const [command, ...args] = this.command
+    const id = crypto.randomBytes(5).toString('hex')
+    const startTime = new Date().toISOString()
+    const spawnResult = spawnSync(command, args, {env: {...process.env, DD_CUSTOM_PARENT_ID: id}, stdio: 'inherit'})
+    const endTime = new Date().toISOString()
+    const exitCode = spawnResult.status ?? this.signalToNumber(spawnResult.signal) ?? 127
+    const api = this.getApiHelper()
+    const [provider, data] = this.getData()
+    const commandStr = this.command.join(' ')
+    await api.reportCustomSpan(
+      {
+        command: commandStr,
+        custom: {
+          id,
+          parent_id: process.env.DD_CUSTOM_PARENT_ID,
+        },
+        data,
+        end_time: endTime,
+        is_error: exitCode !== 0,
+        name: this.name ?? commandStr,
+        start_time: startTime,
+        tags: this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {},
+      },
+      provider
+    )
+
+    return exitCode
+  }
+
+  public getData(): [string, Record<string, string>] {
```

Comment:
`getData` does not tell much of what the function is doing. Maybe `getCIMetadata` ?